llm/src/mosaic_gpt.py

```diff
@@ -186,7 +188,13 @@ def forward(self,
 
         tok_emb = self.transformer.wte(input_ids)  # type: ignore
         pos_emb = self.transformer.wpe(pos)  # type: ignore
-        x = self.transformer.emb_drop(tok_emb + pos_emb)  # type: ignore
+        if self.embedding_fraction == 1:
+            x = self.transformer.emb_drop(tok_emb + pos_emb)  # type: ignore
+        else:
+            x = tok_emb + pos_emb
+            x = self.transformer.emb_drop(
+                x = x * self.embedding_fraction + x.detach() * (1 - self.embedding_fraction)
```

Comment:
Is the variable assignment in the expression a typo?

Reply:
yup. fixed